File: ruby/rubymonk/length_of_strings/spec/length_of_strings_spec.rb
require 'spec_helper'
require_relative '../lib/length_of_strings'

describe '#length_of_strings' do 
  it 'does something' do 
    expect(length_of_strings(['ruby'])).to_not eq(nil)
  end

  it 'provides correct value for array of 1 string' do 
    expect(length_of_strings ['ruby']).to eq([4])
  end

  it 'provides correct value for array of multiple string' do 
    expect(length_of_strings ['ruby', 'foo', 'grill']).to eq([4, 3, 5])
  end
end

File: javascript/recursive/length_of_strings/spec/length_of_strings_spec.js
var lengthOfStrings = require('../lib/length_of_strings');

describe('lengthOfStrings', function() {
  it('is a function', function() {
    expect(typeof lengthOfStrings).toBe('function');
  });

  it('handles simple case', function() {
    expect(lengthOfStrings(['array'])).toBe([1]);
  });
});